src/services/completions.ts

```diff
@@ -5829,3 +5833,31 @@ function toUpperCharCode(charCode: number) {
     }
     return charCode;
 }
+
+/**
+ * These are all the contextual keywords that would be valid to auto-import
+ * in expression space and also a valid keyword in the same location, depending
+ * on what gets typed afterwards. In these cases, we want to offer both the
+ * auto-import and the keyword completion. For example,
+ *
+ * ```ts
+ * type
+ * ```
+ *
+ * may be the beginning of a type alias declaration (keyword completion), or
+ * it may be the beginning of
+ *
+ * ```ts
+ * import { type } from "os";
+ * type() === "Darwin" ? doSomething() : doSomethingElse();
+ * ```
+ */
+function isContextualKeywordInAutoImportableExpressionSpace(keyword: string) {
+    return keyword === "abstract" ||

```

Comment:
I think canonically we'd use a switch for this (faster at runtime I assume).

Reply:
Didn’t we move the visitor API off a `switch` and onto a lookup table for performance reasons?

Reply:
Anyway, this list seems small enough and is unlikely to grow

Reply:
No, that was for a different reason IIRC, the function was so big that it didn't optimize with so many cases handled inside of it